src/interop.cpp

```diff
@@ -0,0 +1,127 @@
+#include "interop.h"
+#include "ConfusableMatcher/ConfusableMatcher.h"
+
+using namespace Napi;
+
+ConfusableMatcherNapiInterop::ConfusableMatcherNapiInterop(const Napi::CallbackInfo &info) : ObjectWrap(info)
+{
+    Napi::Array Map = info[0].As<Napi::Array>();
+    std::vector<std::pair<std::string, std::string>> InputMap;
+    for (uint32_t x = 0; x < Map.Length(); x++)
+    {
+        std::pair<std::string, std::string> kvPair;
+        Napi::Array result = Map.Get(x).As<Napi::Array>();
+        kvPair.first = result.Get((uint32_t)0).As<Napi::String>().Utf8Value();
+        kvPair.second = result.Get((uint32_t)1).As<Napi::String>().Utf8Value();
+        InputMap.push_back(kvPair);
+    }
+
+    Napi::Array Ignores = info[1].As<Napi::Array>();
+    std::unordered_set<std::string> Skips;
+    for (uint32_t x = 0; x < Ignores.Length(); x++)
+    {
+        std::pair<std::string, std::string> kvPair;
+        Napi::String ignore = Map.Get(x).As<Napi::String>();
+        Skips.insert(ignore.Utf8Value());
+    }
+
+    Napi::Boolean AddDefaultValues = info[2].As<Napi::Boolean>();
+    this->_instance = new ConfusableMatcher(InputMap, Skips, AddDefaultValues);
+}
+
+ConfusableMatcherNapiInterop::~ConfusableMatcherNapiInterop()
+{
+    delete this->_instance;
+}
+
+Napi::Value ConfusableMatcherNapiInterop::AddMapping(const Napi::CallbackInfo &info)
+{
+    Napi::Env env = info.Env();
+    Napi::String Key = info[0].As<Napi::String>();
+    Napi::String Value = info[1].As<Napi::String>();
+    Napi::Boolean CheckValueDuplicate = info[2].As<Napi::Boolean>();
+    bool result = this->_instance->AddMapping(Key.Utf8Value(), Value.Utf8Value(), CheckValueDuplicate.ToBoolean());
+    return Napi::Boolean::New(env, result);
+}
+
+Napi::Value ConfusableMatcherNapiInterop::AddSkip(const Napi::CallbackInfo &info)
+{
+    Napi::Env env = info.Env();
+    Napi::String In = info[0].As<Napi::String>();
+    bool result = this->_instance->AddSkip(In.Utf8Value());
```

Comment:
This is not thread safe if called during IndexOf call

Reply:
Removed ref to internal method